pkg/datapath/linux/node.go

```diff
@@ -1446,6 +1470,101 @@ func (n *linuxNodeHandler) NodeNeighborRefresh(ctx context.Context, nodeToRefres
 	}
 }
 
+// NodeCleanNeighbors cleans all neighbor entries for previously used network
+// link interfaces. It should be used when the agent changes the state from
+// `n.enableNeighDiscovery = true` to `n.enableNeighDiscovery = false`.
+func (n *linuxNodeHandler) NodeCleanNeighbors() {
+	devToClean, err := loadNeighLinks(option.Config.StateDir)
+	if err != nil {
+		log.WithError(err).Error("Unable to load old neigh links configuration." +
+			" Previous running state needs to be cleaned up manually.")
+		return
+	}
+	if len(devToClean) == 0 {
+		return
+	}
+	// Delete the file after cleaning up neighbor list.
+	defer os.Remove(filepath.Join(option.Config.StateDir, neighListFileName))
+
+	n.mutex.Lock()
+	defer n.mutex.Unlock()
+
+	for _, linkName := range devToClean {
+		l, err := netlink.LinkByName(linkName)
+		if err != nil {
+			if _, ok := err.(netlink.LinkNotFoundError); !ok {
+				log.WithError(err).WithFields(logrus.Fields{
+					logfields.Device: linkName,
+				}).Error("Unable to GC old neighbor entries for network device")
+			}
+			continue
+		}
+
+		neighList, err := netlink.NeighListExecute(netlink.Ndmsg{
+			Family: netlink.FAMILY_V4,
+			Index:  uint32(l.Attrs().Index),
+			State:  netlink.NUD_PERMANENT,
+		})
+		if err != nil {
+			log.WithError(err).WithFields(logrus.Fields{
+				logfields.Device:    linkName,
+				logfields.LinkIndex: l.Attrs().Index,
+			}).Error("Unable to GC old neighbor entries for network device")
+			continue
+		}
+		for _, neigh := range neighList {
+			err := netlink.NeighDel(&neigh)
```

Comment:
I think we should log an info msg that an ARP entry has been removed.

Reply:
I understand the why but on the other side we could end up with lots of log messages for large clusters whenever cilium-agent initialized.

Reply:
Then maybe "count=N msg=Removed PERM ARP entries previously installed by cilium-agent".